src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/StaffConfigManager.kt

```diff
@@ -58,6 +64,86 @@ class StaffConfigManager(
     }
   }
 
+  fun patchStaffConfiguration(
+    prisonCode: String,
+    staffId: Long,
+    request: PatchStaffConfigRequest,
+  ) {
+    val staffConfig = staffConfigRepository.findByStaffId(staffId) ?: throw EntityNotFoundException("Staff config not found")
```

Comment:
I think this may not cover the staff config ACs properly, particularly 1c. It shouldn't throw an exception if it's not found. Instead if there is no config, it should check if any of status, capacity or reactivateOn are non null and create the relevant config instead.